CIP-NFTMetadataStandard.md

```diff
@@ -0,0 +1,119 @@
+---
+CIP:
+Title: NFT Metadata Standard
+Authors: Alessandro Konrad <alessandro.konrad@live.de>, Smaug <smaug@pool.pm>
+Comments-URI:
+Status: Draft
+Type: Informational
+Created: 2021-04-08
+Post-History: https://forum.cardano.org/t/cip-nft-metadata-standard/45687 and https://www.reddit.com/r/CardanoDevelopers/comments/mkhlv8/nft_metadata_standard/
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This proposal defines an NFT Metadata Standard for Native Tokens.
+
+## Motivation
+
+Tokens on Cardano are a part of the ledger. Unlike on Ethereum, where metadata can be attached to a token through a smart contract, this isn't possible on Cardano because tokens are native and Cardano uses a UTxO ledger, which makes it hard to directly attach metadata to a token.
+So the link to the metadata needs to be established differently.
+Cardano has the ability to send metadata in a transaction, that's the way we can create a link between a token and the metadata. To make the link unique, the metadata should be appended to the same transaction, where the token forge happens:
+
+> Given a token in a EUTXOma ledger, we can ask “where did this token come from?” Since tokens
+> are always created in specific forging operations, we can always trace them back through their
+> transaction graph to their origin.
+
+(Section 4.1 in the paper: https://hydra.iohk.io/build/5400786/download/1/eutxoma.pdf)
+
+## Considerations
+
+That being said, we have unique metadata link to a token and can always prove that with 100% certainty. No one else can manipulate the link except if the policy allows it to ([update mechanism](#update-metadata-link-for-a-specific-token)).
+
+## Specification
+
+This is the registered `transaction_metadatum_label` value
+
+| transaction_metadatum_label | description  |
+| --------------------------- | ------------ |
+| 721                         | NFT Metadata |
+
+### Structure
+
+The structure allows for multiple token mints, also with different policies, in a single transaction.
+
+```
+{
+  "721": {
+    "<policy_id>": {
+      "<asset_name>": {
+        "name": "<name>",
+        "image": "<uri>",
+        "description": "<description>"
+
+        "type": "<mime_type>",
+        "src": "<uri>"
+
+        <other properties>
+      },
+      ...
+    },
+    ...,
+    "version": "1.0"
+  }
+}
+```
+
+The **`image`** and **`name`** property are marked as required. **`image`** should be an URI pointing to a resource with mime type `image/*` used as thumbnail or as actual link if the NFT is an image (ideally <= 1MB).
```

Comment:
There is no reason for *image* property to be required, there are several NFT application which have 0 reason to have any image, even as thumbnail.

Reply:
Of course, but it is fair to assume majority of these applications would require a custom or no metadata at all, therefore this CIP does not applies to them.

Reply:
In such case, if the goal is not to create a generic NFT standard, with reasonable **required** and **optional** fields, it is not wise to call it **NFT Metadata Standard**, but rather **Image NFT Metadata Standard** so it is made clear that this is intended for images only / mostly, as opposed to being a generic NFT attempt to define a NFT standard.

Perhaps this should be two separate CIPs, one for the metadata part that is important for lookup (including a table that lists reserved metadata labels for different NFT applications), and then second CIP that extends each metadata label based off of use-case for the NFT, in this case images.

Reply:
Do you have an example of such a use case where it would need to use this 721 format without needing an image?

Reply:
@mark-stopka this makes sense. The name could be reworded indeed, maybe "Media NFT Metadata Standard"?

Reply:
That's right, there are NFTs that do not require images. But wallets and most 3rd parties will most likely not display or show for example video NFTs or 3D model NFTs,etc. An image on the other side is more universal and gives a great user experience, even tho the main NFT is not an image, but then you see at least a small preview. Not every 3rd party will support exotic types.

Reply:
> That's right, there are NFTs that do not require images. But wallets and most 3rd parties will most likely not display or show for example video NFTs or 3D model NFTs,etc. An image on the other side is more universal and gives a great user experience, even tho the main NFT is not an image, but then you see at least a small preview. Not every 3rd party will support exotic types.

Which should have no impact on this. Wallets should not render images from untrusted sources for security reasons anyway. Also it is important to reflect in an NFT what are you actually buying, are you buying some exclusive rights with the (image) NFT, or just a `sh*tcoin` for giggles? E.q. with vector based NFT's, I can easily imagine, that the buyer would be buying a right to see and create derived work, in which case he will need access to *svg* or similar *source code*.

Which also goes into what @SmaugPool is saying, if your desire is to create generic **Media** NFT standard, then there must be an array of both *payload* and *mime-types*.

Reply:
@mark-stopka about the array, see my edited post where I added a note about the need for multiple additional files:

https://github.com/cardano-foundation/CIPs/pull/85#issuecomment-816540793

Reply:
@mark-stopka this proposal is not only about images, using the `type` and `src` optional fields, it should support as is also:
* audio
* video
* ebook
But all of those benefit from having a thumbnail to represent them (the `image` field).

Reply:
If the intent of the `image` field is to be used as a thumbnail, what about calling it `thumbnail` or `preview` to disambiguate its role?

Reply:
The image field can be used for NFTs that are images. In my opinion most NFTs are images/art. The idea behind thumbnail is that if the the NFT is not an image, you still have a preview and the actual NFT is then linked under the src field. So it's not only meant for thumbnails.

Reply:
Hello Allessandro, I have tried it out:
https://explorer.cardano.org/de/transaction?id=9e680460a65812ea940b69bd271f8a835cfb751de7e39c01fd5b3d072363e3d9
Looks good to me. Base64 encoded datauri allows to store the whole content on blockchain.